models/actions/variable.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package actions
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"code.gitea.io/gitea/models/db"
+	"code.gitea.io/gitea/modules/timeutil"
+	"code.gitea.io/gitea/modules/util"
+
+	"xorm.io/builder"
+)
+
+type ActionVariable struct {
+	ID          int64              `xorm:"pk autoincr"`
+	OwnerID     int64              `xorm:"UNIQUE(owner_repo_title)"`
+	RepoID      int64              `xorm:"UNIQUE(owner_repo_title)"`
```

Comment:
```suggestion
	RepoID      int64              `xorm:"INDEX UNIQUE(owner_repo_title)"`
```

If you have query like `WHERE repo_id=?`, you still need this index.